peer/example_test.go

```diff
@@ -0,0 +1,93 @@
+// Copyright (c) 2015 The btcsuite developers
+// Use of this source code is governed by an ISC
+// license that can be found in the LICENSE file.
+
+package peer_test
+
+import (
+	"errors"
+	"fmt"
+	"log"
+	"net"
+	"time"
+
+	"github.com/btcsuite/btcd/addrmgr"
+	"github.com/btcsuite/btcd/peer"
+	"github.com/btcsuite/btcd/wire"
+)
+
+// zeroHash is the zero value hash (all zeros).  It is defined as a convenience.
+var zeroHash wire.ShaHash
+
+// Not implemented.
+func lookupFunc(host string) ([]net.IP, error) {
+	return nil, errors.New("not implemented")
+}
+
+// newestSha returns the latest known block to this peer.  It returns a
+// hard-coded hash and height since this is only used for example code.
+func newestSha() (*wire.ShaHash, int32, error) {
+	hashStr := "14a0810ac680a3eb3f82edc878cea25ec41d6b790744e5daeef"
+	hash, err := wire.NewShaHashFromStr(hashStr)
+	if err != nil {
+		return nil, 0, err
+	}
+	return hash, 234439, nil
+}
+
+// This example demonstrates initializing both inbound and outbound peers.  An
+// inbound peer listening on mainnet port i.e 8333 is started first, then an
+// outbound peer is connected to it.
+// Peers negotiate protocol by exchanging version and verack messages.  For
+// demonstration, a simple handler for version message is attached to both
+// peers.
+func Example_peerConnection() {
+	addrMgr := addrmgr.New("test", lookupFunc)
+	peerCfg := &peer.Config{
+		UserAgentName:    "peer",
+		UserAgentVersion: "1.0",
+	}
+	listening := make(chan error)
+	go func() {
+		l1, err := net.Listen("tcp", "127.0.0.1:8333")
+		if err != nil {
+			listening <- err
+			return
+		}
+		listening <- nil
+		c1, err := l1.Accept()
+		if err != nil {
+			log.Fatalf("Listen: error %v\n", err)
+		}
+		p1 := peer.NewInboundPeer(newestSha, 0, wire.MainNet, wire.SFNodeNetwork, addrMgr, peerCfg, c1)
```

Comment:
I edited the message.  I noticed you already have the `Config` struct, I'd make use of that.  Perhaps the nonce should remain separate though as it's likely the caller will reuse the config struct and the nonce needs to be unique (random) per peer.